src/t_set.c

```diff
@@ -896,22 +889,24 @@ void spopWithCountCommand(client *c) {
         while (setTypeNext(si, &str, &len, &llele) != -1) {
             if (str == NULL) {
                 addReplyBulkLongLong(c,llele);
-                objele = createStringObjectFromLongLong(llele);
+                propargv[index++] = createStringObjectFromLongLong(llele);
             } else {
                 addReplyBulkCBuffer(c, str, len);
-                objele = createStringObject(str, len);
+                propargv[index++] = createStringObject(str, len);
             }
-
-            /* Replicate/AOF this command as an SREM operation */
-            propargv[2] = objele;
-            alsoPropagate(c->db->id,propargv,3,PROPAGATE_AOF|PROPAGATE_REPL);
-            decrRefCount(objele);
         }
+        /* Replicate/AOF this command as an SREM operation */
+        alsoPropagate(c->db->id,propargv,2 + count,PROPAGATE_AOF|PROPAGATE_REPL);
         setTypeReleaseIterator(si);
 
         /* Assign the new set as the key value. */
         dbReplaceValue(c->db,c->argv[1],newset);
     }
+    for (unsigned long i = 0; i < count; i++) {
```

Comment:
I think I agree with soloestoy that we should probably limit the SREM size, I would suggest to maybe 1024 elements for now. I think it's an easy change, we can move the propagation code here only propagate part of the list at a time. After each propagation we can move the SREM and key pointers forward, so we don't even need a second buffer.